safehttp/plugins/xsrf/xsrf.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xsrf
+
+import (
+	"fmt"
+	"github.com/google/go-safeweb/safehttp"
+
+	// TODO(@empijei, @kele, @mattiasgrenfeldt, @mihalimara22): decide whether
+	// we want to depend on this package or reimplement thefunctionality
+	"golang.org/x/net/xsrftoken"
+)
+
+const (
+	// TokenKey is the form key used when sending the token as part of POST
+	// request.
+	TokenKey = "xsrf-token"
+)
+
+// StorageService  contains information about the users of the web application,
+// including their IDs, needed in generating the XSRF token.
+type StorageService interface {
+	// GetUserID returns the ID of the user making the request based on the
+	// incoming request. If an error occurs, it returns it together with an
+	// empty string.
+	// TODO(@mihalimara22): add a *safehttp.IncomingRequest as a parameter to
+	// this function once the method for this is exported.
+	GetUserID() (string, error)
```

Comment:
I don't get this TODO. What's missing?

Reply:
We need to pass a parameter to the GetUserID function to determine the user. Probably the solution would be to pass a `safehttp.IncomingRequest`, WDYT?

If we choose to pass a `sfehttp.IncomingRequest`, I can't test it until we submit the PR where `safehttp.NewIncomingRequest` is exported because I can't create safe requests. That's why I left a TODO here and in the test file.

Reply:
+1 on passing the request as parameter

Reply:
Wait, why do you need `safehttp.NewIncomingRequest`?

Reply:
The `userIDStorage` needs some information to determine what `userID` to return.